<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Multimodes of pathwAY Activation • MAYA</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Multimodes of pathwAY Activation">
<meta property="og:description" content="Unsupervised cell type annotation and multimodal pathway activation study">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">MAYA</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/MAYA_user_guide.html">MAYA User Guide</a>
    </li>
  </ul>
</li>
<li>
  <a href="news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="contents col-md-9">

<div class="section level1">
<div class="page-header"><h1 id="maya-user-guide">MAYA User Guide<a class="anchor" aria-label="anchor" href="#maya-user-guide"></a>
</h1></div>
<p>YL and CV for OBS</p>
</div>
<div class="section level1">
<h1 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h1>
<p>MAYA (Multimodes of pathwAY Activation) enables comprehensive pathway study and automated cell type prediction thanks to multimodal activity scoring of gene lists (pathways from MSigDB or markers from PanglaoDB for instance) in individual cells.</p>
<p>MAYA provides functions to build a multimodal pathway activity matrix, compute UMAP to visualize cells in 2 dimensions based on this activity matrix, visualize the different modes of activation of a pathway in this space and also check the expression of top contributing genes of each mode in the dataset through heatmaps.</p>
<p>MAYA also provides a function to predict cell type based on PanglaoDB or any custom cell type markers list provided by the user. Cells can also be visualized in embeddings computed based on the marker activity matrix. MAYA is not run in a multimodal way for cell type prediction for interpretability purposes.</p>
</div>
<div class="section level1">
<h1 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h1>
<p>This package requires R &gt;= 4.0.5 to run and can be installed using devtools:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"one-biosciences/maya"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level1">
<h1 id="use-case-kidney-dataset">Use case: kidney dataset<a class="anchor" aria-label="anchor" href="#use-case-kidney-dataset"></a>
</h1>
<p>Through this use case, we will demonstrate the insight that brings multimodal pathway activity analysis to apprehend the cellular heterogeneity of a dataset. We will show how to use the different functions available in this package to run a full analysis of the activity of MSigDB HALLMARK gene sets in normal kidney cells. We will also demonstrate how to use the integrated MAYA function to automatically predict cell types.</p>
<p>This example dataset was built using a count matrix and corresponding metadata made available by <em>Young et al.(2018)</em>. In their paper, the authors thoroughly investigated the cellular identity of normal and cancerous human kidneys from 72,501 single-cell transcriptomes, which led to the identification of various cell types and subtypes of normal and cancerous immune and non-immune cell types.</p>
<p>The matrix was filtered to select only protein coding genes and keep only 5 groups of <strong>normal</strong> cells:</p>
<ul>
<li>
<strong>AV2</strong> (Vascular endothelium – ascending vasa recta)</li>
<li>
<strong>G</strong> (Glomerular epithelium - podocytes)</li>
<li>
<strong>M</strong> (Mesangial cells - myofibroblasts)</li>
<li>
<strong>8T</strong> (CD8 T cells)</li>
<li>
<strong>MNP1</strong> (Mononuclear phagocytes)</li>
</ul>
<div class="section level2">
<h2 id="loading-data">Loading data<a class="anchor" aria-label="anchor" href="#loading-data"></a>
</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">MAYA</span><span class="op">)</span></span></code></pre></div>
<p>First, load the example count matrix available in the package:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">path</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"kidney_count_matrix.tsv"</span>, package <span class="op">=</span> <span class="st">"MAYA"</span><span class="op">)</span></span>
<span><span class="va">count_mat</span><span class="op">&lt;-</span><span class="fu">Matrix</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/Matrix/man/Matrix.html" class="external-link">Matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span>file<span class="op">=</span><span class="va">path</span>,sep<span class="op">=</span><span class="st">"\t"</span>,header<span class="op">=</span><span class="cn">T</span>,check.names <span class="op">=</span> <span class="cn">F</span><span class="op">)</span><span class="op">)</span>,sparse<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">count_mat</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="do">## [1] 19720  1252</span></span></code></pre></div>
<p>The number of cells was intentionally chosen small to ensure reasonable running time for this example.</p>
<p>Then, load the associated metadata:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">path</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"kidney_metadata.tsv"</span>, package <span class="op">=</span> <span class="st">"MAYA"</span><span class="op">)</span></span>
<span><span class="va">meta</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span>file<span class="op">=</span><span class="va">path</span>,sep<span class="op">=</span><span class="st">"\t"</span>,header<span class="op">=</span><span class="cn">T</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">meta</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb6"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="do">## [1] 1252    5</span></span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">meta</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb8"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a><span class="do">## [1] "nCount_RNA"   "nFeature_RNA" "Alias"        "Cell_type1"   "Cell_type2"</span></span></code></pre></div>
<p>We have several variables available to characterize our cells:</p>
<ul>
<li>“nCount_RNA” (number of molecules counted)</li>
<li>“nFeature_RNA” (number of genes expressed)</li>
<li>“Alias” (short cell subtype name)</li>
<li>“Cell_type1” (cell type description)</li>
<li>“Cell_type2” (cell subtype description)</li>
</ul>
<p>“Alias” is the annotation that will be preferred to describe our cells as it corresponds to the annotation of the 5 groups described above.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">meta</span><span class="op">$</span><span class="va">Alias</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb10"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb10-2"><a href="#cb10-2" tabindex="-1"></a><span class="do">##   8T  AV2    G    M MNP1 </span></span>
<span id="cb10-3"><a href="#cb10-3" tabindex="-1"></a><span class="do">##  155  305  259  118  415</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="running-maya_pathway_analysis">Running MAYA_pathway_analysis<a class="anchor" aria-label="anchor" href="#running-maya_pathway_analysis"></a>
</h2>
<p>The user simply needs a raw count matrix and to specify in <em>modules_list</em> either “hallmark” or “kegg” to load prebuilt versions of the corresponding MSigDB pathway lists, or provide any list of pathways with their associated markers (they can be loaded from gmt files with the function MAYA::read_gmt).</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">activity_summary</span><span class="op">&lt;-</span><span class="fu"><a href="reference/MAYA_pathway_analysis.html">MAYA_pathway_analysis</a></span><span class="op">(</span>expr_mat<span class="op">=</span><span class="va">count_mat</span>,</span>
<span>                                        modules_list <span class="op">=</span> <span class="st">"hallmark"</span>,</span>
<span>                                        is_logcpm<span class="op">=</span><span class="cn">F</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb12"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a><span class="do">## Running pathway analysis</span></span>
<span id="cb12-2"><a href="#cb12-2" tabindex="-1"></a></span>
<span id="cb12-3"><a href="#cb12-3" tabindex="-1"></a><span class="do">## Loading HALLMARK from MSigDB</span></span>
<span id="cb12-4"><a href="#cb12-4" tabindex="-1"></a></span>
<span id="cb12-5"><a href="#cb12-5" tabindex="-1"></a><span class="do">## Computing gene sets activity</span></span>
<span id="cb12-6"><a href="#cb12-6" tabindex="-1"></a></span>
<span id="cb12-7"><a href="#cb12-7" tabindex="-1"></a><span class="do">## Found at least one informative activation mode for 20 gene sets</span></span>
<span id="cb12-8"><a href="#cb12-8" tabindex="-1"></a></span>
<span id="cb12-9"><a href="#cb12-9" tabindex="-1"></a><span class="do">## 'as(&lt;dsCMatrix&gt;, "dgTMatrix")' is deprecated.</span></span>
<span id="cb12-10"><a href="#cb12-10" tabindex="-1"></a><span class="do">## Use 'as(as(., "generalMatrix"), "TsparseMatrix")' instead.</span></span>
<span id="cb12-11"><a href="#cb12-11" tabindex="-1"></a><span class="do">## See help("Deprecated") and help("Matrix-deprecated").</span></span></code></pre></div>
<p>The result is a list containing the main fields:</p>
<ul>
<li>activity_matrix: mode by cell matrix</li>
<li>umap: activity-based embeddings for result exploration</li>
<li>PCA_obj: detailed result of activity analysis by pathway, required for some plotting functions</li>
</ul>
<p>The user can then visualize the activity matrix as a heatmap:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/plot_heatmap_activity_mat.html">plot_heatmap_activity_mat</a></span><span class="op">(</span>activity_mat <span class="op">=</span> <span class="va">activity_summary</span><span class="op">$</span><span class="va">activity_matrix</span>, </span>
<span>                          meta <span class="op">=</span> <span class="va">meta</span>, </span>
<span>                          annot_name <span class="op">=</span> <span class="st">"Alias"</span><span class="op">)</span></span></code></pre></div>
<p><img src="README_figs/README-unnamed-chunk-9-1.png"><!-- --></p>
<p>The user can also choose to scale each mode between 0 and 1, which sometimes help homogenize the visualization when some modes have high activation compared with others.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/plot_heatmap_activity_mat.html">plot_heatmap_activity_mat</a></span><span class="op">(</span>activity_mat <span class="op">=</span> <span class="fu"><a href="reference/scale_0_1.html">scale_0_1</a></span><span class="op">(</span><span class="va">activity_summary</span><span class="op">$</span><span class="va">activity_matrix</span><span class="op">)</span>, </span>
<span>                          meta <span class="op">=</span> <span class="va">meta</span>, </span>
<span>                          annot_name <span class="op">=</span> <span class="st">"Alias"</span><span class="op">)</span></span></code></pre></div>
<p><img src="README_figs/README-unnamed-chunk-10-1.png"><!-- --></p>
<p>The user can also check how cells from different cell types organize in MAYA embeddings based on hallmark pathways activity:</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/plot_umap_annot.html">plot_umap_annot</a></span><span class="op">(</span>umap<span class="op">=</span><span class="va">activity_summary</span><span class="op">$</span><span class="va">umap</span>,</span>
<span>                labels <span class="op">=</span> <span class="va">meta</span><span class="op">$</span><span class="va">Alias</span>,</span>
<span>                title <span class="op">=</span> <span class="st">"Author annotation - HALLMARK"</span><span class="op">)</span></span></code></pre></div>
<p><img src="README_figs/README-unnamed-chunk-11-1.png"><!-- --></p>
<p>The user can then visualize multimodal activity of a specific pathway on the UMAP - the allograft rejection for instance - and check the expression of top 10 contributing genes for each mode. For this purpose, the user might want to plot the logCPM counts and therefore normalize the matrix at this stage.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/plot_umap_pathway_activity.html">plot_umap_pathway_activity</a></span><span class="op">(</span>umap<span class="op">=</span><span class="va">activity_summary</span><span class="op">$</span><span class="va">umap</span>,</span>
<span>                           PCA_object <span class="op">=</span> <span class="va">activity_summary</span><span class="op">$</span><span class="va">PCA_obj</span>,</span>
<span>                           module <span class="op">=</span> <span class="st">"HALLMARK_ALLOGRAFT_REJECTION"</span><span class="op">)</span></span></code></pre></div>
<p><img src="README_figs/README-unnamed-chunk-12-1.png"><!-- --></p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">logcpm</span><span class="op">&lt;-</span><span class="fu"><a href="reference/logcpmNormalization.html">logcpmNormalization</a></span><span class="op">(</span><span class="va">count_mat</span><span class="op">)</span></span>
<span><span class="fu"><a href="reference/plot_heatmap_pathway_top_contrib_genes.html">plot_heatmap_pathway_top_contrib_genes</a></span><span class="op">(</span>expr_mat<span class="op">=</span><span class="va">logcpm</span>,</span>
<span>                                       PCA_object <span class="op">=</span> <span class="va">activity_summary</span><span class="op">$</span><span class="va">PCA_obj</span>,</span>
<span>                                       module <span class="op">=</span> <span class="st">"HALLMARK_ALLOGRAFT_REJECTION"</span>,</span>
<span>                                       n <span class="op">=</span> <span class="fl">10</span>,</span>
<span>                                       meta <span class="op">=</span> <span class="va">meta</span>,</span>
<span>                                       annot_name <span class="op">=</span> <span class="st">"Alias"</span><span class="op">)</span></span></code></pre></div>
<p><img src="README_figs/README-unnamed-chunk-13-1.png"><!-- --></p>
<p>The expression of top contributing genes can also be plotted on the UMAP.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/plot_umap_gene.html">plot_umap_gene</a></span><span class="op">(</span>umap<span class="op">=</span><span class="va">activity_summary</span><span class="op">$</span><span class="va">umap</span>,</span>
<span>               expr_mat <span class="op">=</span> <span class="va">logcpm</span>,</span>
<span>               gene <span class="op">=</span> <span class="st">"CTSS"</span><span class="op">)</span></span></code></pre></div>
<p><img src="README_figs/README-unnamed-chunk-14-1.png"><!-- --></p>
<p>Finally, the user can compute and represent the specificity of each mode in the different cell types.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/plot_pathway_specificity.html">plot_pathway_specificity</a></span><span class="op">(</span>PCA_object <span class="op">=</span> <span class="va">activity_summary</span><span class="op">$</span><span class="va">PCA_obj</span>,</span>
<span>                         module <span class="op">=</span> <span class="st">"HALLMARK_ALLOGRAFT_REJECTION"</span>,</span>
<span>                         meta <span class="op">=</span> <span class="va">meta</span>,</span>
<span>                         annot_name <span class="op">=</span> <span class="st">"Alias"</span><span class="op">)</span></span></code></pre></div>
<p><img src="README_figs/README-unnamed-chunk-15-1.png"><!-- --></p>
</div>
<div class="section level2">
<h2 id="running-maya_predict_cell_type">Running MAYA_predict_cell_type<a class="anchor" aria-label="anchor" href="#running-maya_predict_cell_type"></a>
</h2>
<p>To run the cell type prediction module in the simplest way, one only needs a raw count matrix, and the algorithm will load the full PanglaoDB to try to identify cell types. However some parameters might be important to have in mind like <em>min_cells_pct</em> which corresponds to the minimal fraction of cells you expect a cell type to represent in the whole dataset (default is 5%), <em>organs</em> to directly specify the tissue you are working on to avoid possible interferences between similar cell types located in different organs, <em>is_logcpm</em> that should be set to true if one already performed a custom data normalization, <em>nCores</em> to speed up the process if possible and <em>compute_umap</em> to disable this option that can take more time as the size of the dataset increases.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># activity_summary&lt;-MAYA_predict_cell_types(expr_mat = count_mat)</span></span>
<span><span class="va">activity_summary</span><span class="op">&lt;-</span><span class="fu"><a href="reference/MAYA_predict_cell_types.html">MAYA_predict_cell_types</a></span><span class="op">(</span>expr_mat <span class="op">=</span> <span class="va">count_mat</span>,</span>
<span>                                    min_cells_pct <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>                                    organs <span class="op">=</span> <span class="st">"Kidney"</span>,</span>
<span>                                    is_logcpm <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                                    nCores <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                                    compute_umap <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb21"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" tabindex="-1"></a><span class="do">## Running cell type identification</span></span>
<span id="cb21-2"><a href="#cb21-2" tabindex="-1"></a></span>
<span id="cb21-3"><a href="#cb21-3" tabindex="-1"></a><span class="do">## Loading PanglaoDB</span></span>
<span id="cb21-4"><a href="#cb21-4" tabindex="-1"></a></span>
<span id="cb21-5"><a href="#cb21-5" tabindex="-1"></a><span class="do">## Computing gene sets activity</span></span>
<span id="cb21-6"><a href="#cb21-6" tabindex="-1"></a></span>
<span id="cb21-7"><a href="#cb21-7" tabindex="-1"></a><span class="do">## Found at least one informative activation mode for 35 gene sets</span></span>
<span id="cb21-8"><a href="#cb21-8" tabindex="-1"></a></span>
<span id="cb21-9"><a href="#cb21-9" tabindex="-1"></a><span class="do">## Computing UMAP on activity matrix</span></span></code></pre></div>
<p><img src="README_figs/README-unnamed-chunk-16-1.png"><!-- --></p>
<p>The result is a list containing the main fields:</p>
<ul>
<li>cell_annotation: vector with cell type for each cell or “unassigned”</li>
<li>activity_matrix: mode by cell matrix</li>
<li>umap: activity-based embeddings for result exploration</li>
</ul>
<p>The user can visualize cell annotation in MAYA embeddings, compare it with the authors annotation and visualize the underlying activity matrix.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/plot_umap_annot.html">plot_umap_annot</a></span><span class="op">(</span>umap<span class="op">=</span><span class="va">activity_summary</span><span class="op">$</span><span class="va">umap</span>,</span>
<span>                labels <span class="op">=</span> <span class="va">activity_summary</span><span class="op">$</span><span class="va">cell_annotation</span>,</span>
<span>                title <span class="op">=</span> <span class="st">"MAYA cell type prediction"</span><span class="op">)</span></span></code></pre></div>
<p><img src="README_figs/README-unnamed-chunk-17-1.png"><!-- --></p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/plot_umap_annot.html">plot_umap_annot</a></span><span class="op">(</span><span class="va">activity_summary</span><span class="op">$</span><span class="va">umap</span>,labels <span class="op">=</span> <span class="va">meta</span><span class="op">$</span><span class="va">Alias</span>,title <span class="op">=</span> <span class="st">"Author annotation"</span><span class="op">)</span></span></code></pre></div>
<p><img src="README_figs/README-unnamed-chunk-18-1.png"><!-- --></p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/plot_heatmap_activity_mat.html">plot_heatmap_activity_mat</a></span><span class="op">(</span>activity_mat <span class="op">=</span> <span class="va">activity_summary</span><span class="op">$</span><span class="va">activity_matrix</span>, </span>
<span>                          meta <span class="op">=</span> <span class="va">meta</span>, </span>
<span>                          annot_name <span class="op">=</span> <span class="st">"Alias"</span><span class="op">)</span></span></code></pre></div>
<p><img src="README_figs/README-unnamed-chunk-19-1.png"><!-- --></p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb26"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" tabindex="-1"></a><span class="do">## R version 4.4.1 (2024-06-14)</span></span>
<span id="cb26-2"><a href="#cb26-2" tabindex="-1"></a><span class="do">## Platform: x86_64-pc-linux-gnu</span></span>
<span id="cb26-3"><a href="#cb26-3" tabindex="-1"></a><span class="do">## Running under: Ubuntu 22.04.4 LTS</span></span>
<span id="cb26-4"><a href="#cb26-4" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb26-5"><a href="#cb26-5" tabindex="-1"></a><span class="do">## Matrix products: default</span></span>
<span id="cb26-6"><a href="#cb26-6" tabindex="-1"></a><span class="do">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span id="cb26-7"><a href="#cb26-7" tabindex="-1"></a><span class="do">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span id="cb26-8"><a href="#cb26-8" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb26-9"><a href="#cb26-9" tabindex="-1"></a><span class="do">## locale:</span></span>
<span id="cb26-10"><a href="#cb26-10" tabindex="-1"></a><span class="do">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span id="cb26-11"><a href="#cb26-11" tabindex="-1"></a><span class="do">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span id="cb26-12"><a href="#cb26-12" tabindex="-1"></a><span class="do">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span id="cb26-13"><a href="#cb26-13" tabindex="-1"></a><span class="do">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span id="cb26-14"><a href="#cb26-14" tabindex="-1"></a><span class="do">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span id="cb26-15"><a href="#cb26-15" tabindex="-1"></a><span class="do">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span id="cb26-16"><a href="#cb26-16" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb26-17"><a href="#cb26-17" tabindex="-1"></a><span class="do">## time zone: UTC</span></span>
<span id="cb26-18"><a href="#cb26-18" tabindex="-1"></a><span class="do">## tzcode source: system (glibc)</span></span>
<span id="cb26-19"><a href="#cb26-19" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb26-20"><a href="#cb26-20" tabindex="-1"></a><span class="do">## attached base packages:</span></span>
<span id="cb26-21"><a href="#cb26-21" tabindex="-1"></a><span class="do">## [1] parallel  stats     graphics  grDevices utils     datasets  methods  </span></span>
<span id="cb26-22"><a href="#cb26-22" tabindex="-1"></a><span class="do">## [8] base     </span></span>
<span id="cb26-23"><a href="#cb26-23" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb26-24"><a href="#cb26-24" tabindex="-1"></a><span class="do">## other attached packages:</span></span>
<span id="cb26-25"><a href="#cb26-25" tabindex="-1"></a><span class="do">##  [1] MAYA_1.0.1        leidenbase_0.1.30 viridis_0.6.5     viridisLite_0.4.2</span></span>
<span id="cb26-26"><a href="#cb26-26" tabindex="-1"></a><span class="do">##  [5] wesanderson_0.3.7 pheatmap_1.0.12   RANN_2.6.1        igraph_2.0.3     </span></span>
<span id="cb26-27"><a href="#cb26-27" tabindex="-1"></a><span class="do">##  [9] umap_0.2.10.0     ggplot2_3.5.1     dplyr_1.1.4       Matrix_1.7-0     </span></span>
<span id="cb26-28"><a href="#cb26-28" tabindex="-1"></a><span class="do">## [13] rmarkdown_2.28   </span></span>
<span id="cb26-29"><a href="#cb26-29" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb26-30"><a href="#cb26-30" tabindex="-1"></a><span class="do">## loaded via a namespace (and not attached):</span></span>
<span id="cb26-31"><a href="#cb26-31" tabindex="-1"></a><span class="do">##  [1] gtable_0.3.5       jsonlite_1.8.8     highr_0.11         compiler_4.4.1    </span></span>
<span id="cb26-32"><a href="#cb26-32" tabindex="-1"></a><span class="do">##  [5] tidyselect_1.2.1   Rcpp_1.0.13        gridExtra_2.3      scales_1.3.0      </span></span>
<span id="cb26-33"><a href="#cb26-33" tabindex="-1"></a><span class="do">##  [9] png_0.1-8          yaml_2.3.10        fastmap_1.2.0      reticulate_1.38.0 </span></span>
<span id="cb26-34"><a href="#cb26-34" tabindex="-1"></a><span class="do">## [13] lattice_0.22-6     R6_2.5.1           labeling_0.4.3     generics_0.1.3    </span></span>
<span id="cb26-35"><a href="#cb26-35" tabindex="-1"></a><span class="do">## [17] knitr_1.48         tibble_3.2.1       munsell_0.5.1      openssl_2.2.1     </span></span>
<span id="cb26-36"><a href="#cb26-36" tabindex="-1"></a><span class="do">## [21] RColorBrewer_1.1-3 pillar_1.9.0       rlang_1.1.4        utf8_1.2.4        </span></span>
<span id="cb26-37"><a href="#cb26-37" tabindex="-1"></a><span class="do">## [25] xfun_0.47          cli_3.6.3          withr_3.0.1        magrittr_2.0.3    </span></span>
<span id="cb26-38"><a href="#cb26-38" tabindex="-1"></a><span class="do">## [29] digest_0.6.37      grid_4.4.1         askpass_1.2.0      lifecycle_1.0.4   </span></span>
<span id="cb26-39"><a href="#cb26-39" tabindex="-1"></a><span class="do">## [33] vctrs_0.6.5        RSpectra_0.16-2    evaluate_0.24.0    glue_1.7.0        </span></span>
<span id="cb26-40"><a href="#cb26-40" tabindex="-1"></a><span class="do">## [37] farver_2.1.2       fansi_1.0.6        colorspace_2.1-1   tools_4.4.1       </span></span>
<span id="cb26-41"><a href="#cb26-41" tabindex="-1"></a><span class="do">## [41] pkgconfig_2.0.3    htmltools_0.5.8.1</span></span></code></pre></div>
</div>
</div>

  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li>GNU Affero General Public License v3.0</li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing MAYA</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li> <br><small class="roles"> Maintainer </small>  </li>
</ul>
</div>



  </div>
</div>


      <footer><div class="copyright">
  <p></p>
<p>Developed by .</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

      </footer>
</div>






  </body>
</html>
